src/tools/rbd_mirror/InstanceWatcher.h

```diff
@@ -15,6 +15,9 @@
 #include "librbd/managed_lock/Types.h"
 #include "tools/rbd_mirror/instance_watcher/Types.h"
 
+#pragma GCC diagnostic push
```

Comment:
Why not fix the issue instead of turning off the warning for this unit?

Reply:
I do have to claim ignorance of these since my Fedora 26 environment doesn't show these warnings.

Reply:
I would be happy to, though I had thought it was intentional. The problem is that unregister_watch in the base class takes a Context* argument and in this InstanceWater.h class it takes no parameters since it uses one from elsewhere. We could either change the name to something else and use a static_assert/template trick to 'remove' direct calls to the function in the derived class, or add the paremter. (I don't like that idea, since it seems error-prone to take a parameter we don't do anything with and just pass in nullptr or something. Unless there's some obvious use this function could make of a context in addition to the stored one.)

Reply:
What compiler version or cmake settings are you using to see this warning?

Reply:
Clang 4. I thought GCC 6/7 showed it, too, but looking again I'm not seeing it.